app/controllers/documents_v2_controller.rb

```diff
@@ -0,0 +1,119 @@
+class DocumentsV2Controller < ApplicationController
+
+  before_filter :require_valid_access_key, :only => [:open]
+  before_filter :require_valid_read_write_access_key, :only => [:save, :patch]
+  before_filter :require_valid_shared_recordid, :only => [:copy_shared]
+
+  # don't check for csrf token
+  skip_before_filter :verify_authenticity_token, :only => [:save, :patch, :copy_shared]
+
+  # v2 of the api is all anonymous access
+  skip_authorization_check
+
+  include DocumentsV2Helper
+
+  def open
```

Comment:
Because this is the only place that require_valid_access_key is used. I think it would be better to remove the method and before_filter and just inline the code here.  This removes the need for the `@document` instance variable. And the readOnly flag can be set right in the conditional.